book/3-classification.tex

```diff
@@ -268,4 +268,74 @@ \subsection{Phi Coefficient}
 \clearpage
 \thispagestyle{classificationstyle}
 \section{MCC}
-\subsection{Matthew's Correlation Coefficient}
\ No newline at end of file
+\subsection{Matthew's Correlation Coefficient}
+
+
+% ---------- EC ----------
+\clearpage
+\thispagestyle{classificationstyle}
+\section{EC}
+\subsection{Expected Cost}
+
+The Expected Cost (EC) metric is designed to evaluate the performance of decision-making systems. It is the expected value of a cost function that assigns a penalty to each combination of the system's decision and the true class of a sample.
+
+\begin{center}
+\tikz{
+\node[inner sep=2pt, font=\Large] (a) {
+{
+$\displaystyle
+EC = \frac{1}{{\color{red}N}} \sum_{t=1}^{N} {\color{purple}C}({\color{cyan}h_t}, {\color{orange}d_t})
+$
+}
+};
+\draw[-latex,red, semithick] ($(a.south)+(-0.7,0.2)$) to[bend left=30] node[pos=1, left] {number of samples} +(-0.8,-0.5); % Increased length
+\draw[-latex,cyan, semithick] ($(a.north)+(1.5,-0.1)$) to[bend left=30] node[pos=1, right] {True class} +(0.8,0.5); % Increased length
+% \draw[-latex,cyan, semithick] ($(a.south)+(1.2,0.3)$) to[bend left=30] node[pos=1, left] {True class} +(-0.5,-0.5); % Increased length
+\draw[-latex,orange, semithick] ($(a.south)+(2.0,0.2)$) to[bend right=30] node[pos=1, right] {Decision} +(2.0,-0.5); % Increased length
+}
+\end{center}
+
+
+
+For each possible true class $h$ and decision $d$, a cost $C(h, d)$ is assigned. The expected cost (EC) is then computed by averaging the costs over the test samples. The cost depends on the model's decision for each sample and whether the prediction matches the true label.
+
+
+\textbf{When to use EC?}
+
+
+EC allows assigning specific costs to each combination of true class and predicted decision, making it ideal for applications like finance and healthcare where misclassifications have different consequences. It is highly customizable and effectively handles class imbalance.
+
+\coloredboxes{
+\item It can adjust for deployment scenarios by modifying class priors and handling class imbalance.
+\item It is flexible and explicitly allows to encode the relative costs of different errors in a transparent manner.
+
+\item Does not limit the possible decisions to only the class labels.
+}
+{
+\item It's effectiveness depends on well-calibrated model probabilities 
+\item Defining an appropriate cost matrix can be challenging and requires knowledge of the application 
+\item EC is less widely adopted, making it harder to compare results with existing benchmarks.
+}
+
+
+\clearpage
+\thispagestyle{classificationstyle}
+
+\textbf{Understand with an example}
```

Comment:
This is good, but I would like us to stick to the current format of the previous metrics. The second page should visualize the metric under different circumstances, and include an explanation, it should also have the content block "did you know that..." and the other related metrics section.

It is okay if you need an extra paragraph at the beginning of that to explain the metric further.

Reply:
Sure I will add them.

Reply:
Giving it more thought, the table is an excellent addition to this metric. I would get rid of the bold text "Understand with an example" and modify the example in the table, I think a monetary example would be nicer because then the cost matrix is really about costs.

If we keep the table then we might not need a plot for this one.

Here is an example of a similar use case that uses expected cost matrix: https://www.linkedin.com/posts/wiljan-cools_how-can-you-quantify-the-business-value-of-activity-7236285498321764354-pi1X

Reply:
That's great, I will work on it 😀.